### PR TITLE
[#11572] Notification Feature - PUT notification route

### PR DESCRIPTION
**Pull request by NicolasCwy**
Part of  #11572 

To be merged after #11651 and #11660

**Outline of Solution**
Add PUT notification route to update notifications.

### CODE REVIEW

**Comment by fsgmhoward** [comment]
Reviews would be hard at this stage as there are other PRs rebased into this one. They have to be merged before the review of this PR can proceed. Please indicate the list of PRs preceding this one at the description above. @NicolasCwy

**Comment by jianhandev** [comment]
This PR seem to contain a lot of changes from other PRs, is it possible to remove those and only keep ones related to the PUT notification endpoint?

File: src/main/java/teammates/ui/request/NotificationBasicRequest.java [hunk 1]
```diff
@@ -4,9 +4,9 @@
 import teammates.common.datatransfer.NotificationType;
 
 /**
- * The request for creating a notification.
+ * The basic request for a notification.
  */
-public class CreateNotificationRequest extends BasicRequest {
+public class NotificationBasicRequest extends BasicRequest {
     private Long startTimestamp;
     private Long endTimestamp;
```
Comment (jianhandev) [inline src/main/java/teammates/ui/request/NotificationBasicRequest.java:11]: Should these be `long` instead

**Review by jianhandev — COMMENTED** [review]
Looks good!

File: src/main/java/teammates/ui/request/NotificationBasicRequest.java (hunk 1 above)
Comment (NicolasCwy) [reply src/main/java/teammates/ui/request/NotificationBasicRequest.java]: Fixed in the latest commit. Must have changed during the rebase

**Review by fsgmhoward — APPROVED** [review]
LGTM.

**Review by jianhandev — APPROVED** [review]

**Pull request merged**